pkg/kvstore/etcd.go

```diff
@@ -544,8 +544,21 @@ func (e *etcdClient) renewSession(ctx context.Context) error {
 }
 
 func (e *etcdClient) renewLockSession(ctx context.Context) error {
-	<-e.firstSession
-	<-e.lockSession.Done()
+	select {
+	// wait for initial session to be established
+	case <-e.firstSession:
+	// controller has stopped or etcd client is closing
+	case <-ctx.Done():
+		return nil
```

Comment:
Shouldn't be `return ctx.Err()`?

Reply:
Depends, the context is cancelled when the controller stops or when the etcd client is shut down. So `nil` is returned to not report a controller failure if the controller stopped.

Reply:
Note that a later commit changes this to return the `err` after `firstSession` started returning the error observed.